### PR TITLE
Extraction of keyboard handling

### PR DESCRIPTION
**Pull request by jrcmramos-bumble**

### CODE REVIEW

File: Chatto/sources/ChatController/BaseChatViewController.swift [hunk 1]
```diff
@@ -107,6 +108,16 @@ open class BaseChatViewController: UIViewController,
         didSet { self.updateInputContainerBottomConstraint() }
     }
 
+    private func makeKeyboardUpdatesHandler() -> KeyboardUpdatesHandler {
```
Comment (wiruzx) [inline Chatto/sources/ChatController/BaseChatViewController.swift:111]: What do you think about passing `KeyboardUpdatesHandler` as a dependency?

It seems like `BaseChatViewController` has property `keyboardEventsHandlers` only to notify them whenever `KeyboardUpdatesHandler` triggers an even.

If `KeyboardUpdatesHandler` called those `keyboardEventsHandlers` directly, we could remove both `keyboardTracker` and `keyboardEventsHandlers` from the `BaseChatViewController`

File: Chatto/sources/Keyboard/KeyboardUpdatesHandler.swift [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+//
+import UIKit
+
+public protocol KeyboardInputAdjustableViewController: UIViewController {
+    var inputBarContainer: UIView { get }
+    var inputContainerBottomConstraint: NSLayoutConstraint { get }
+}
+
+public protocol KeyboardUpdatesHandlerProtocol {
+    var keyboardTrackingView: KeyboardTrackingView { get }
```
Comment (wiruzx) [inline Chatto/sources/Keyboard/KeyboardUpdatesHandler.swift:12]: I'd suggest to replace `keyboardTrackingView`'s type to UIView and make `KeyboardTrackingView` public

File: Chatto/sources/Keyboard/KeyboardTracker.swift [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+    public init(frame: CGRect,
+                state: KeyboardState) {
+        self.frame = frame
+        self.state = state
+    }
+}
+
+public protocol KeyboardTrackerDelegate: AnyObject {
+    func keyboardTracker(_ : KeyboardTrackerProtocol,
```
Comment (wiruzx) [inline Chatto/sources/Keyboard/KeyboardTracker.swift:26]: What do you think about `func didUpdate(keyboardStatus: KeyboardStatus)` ?

File: Chatto/sources/Keyboard/StaticKeyboardUpdatesAnimator.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
+//
```
Comment (wiruzx) [inline Chatto/sources/Keyboard/StaticKeyboardUpdatesAnimator.swift:1]: I wasn't able to find any usages of this class and protocol in this PR. Should we keep it?

File: Chatto/sources/Keyboard/KeyboardUpdatesHandler.swift [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+    func adjustLayoutIfNeeded()
+    func startTracking()
+    func stopTracking()
+}
+
+public protocol KeyboardUpdatesHandlerDelegate: AnyObject {
+    func keyboardUpdatesHandler(_: KeyboardUpdatesHandler, didAdjustBottomMarginTo bottomMargin: CGFloat)
```
Comment (wiruzx) [inline Chatto/sources/Keyboard/KeyboardUpdatesHandler.swift:20]: I think we're declaring concrete implementation of `KeyboardUpdatesHandler` here.
I'd remove this parameter completely add rename it to:
`func didAdjustBottomMargin(to margin: CGFloat)`

File: Chatto/sources/Keyboard/StaticKeyboardUpdatesAnimator.swift [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+public protocol KeyboardUpdatesAnimatorProtocol {
+    func configure(for KeyboardInputAdjustableViewController: KeyboardInputAdjustableViewController)
+    func adjustInputContentBottomMarginTo(_ newValue: CGFloat)
+}
+
+public final class StaticKeyboardUpdatesAnimator: KeyboardUpdatesAnimatorProtocol {
+
+    private let inputContainerBottomBaseOffset: CGFloat
+
+    private weak var KeyboardInputAdjustableViewController: KeyboardInputAdjustableViewController?
```
Comment (wiruzx) [inline Chatto/sources/Keyboard/StaticKeyboardUpdatesAnimator.swift:15]: lowercase?

**Review by wiruzx — CHANGES_REQUESTED** [review]

File: Chatto/sources/ChatController/BaseChatViewController.swift (hunk 1 above)
Comment (jrcmramos-bumble) [reply Chatto/sources/ChatController/BaseChatViewController.swift:111]: That's a good point! Even if `BaseChatViewController` still needs to add itself as the `keyboardInputAdjustableViewController` and `delegate` we can definitely do that!

**Pull request merged**